x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/CIDRMatch.java

```diff
@@ -101,14 +82,81 @@ protected TypeResolution resolveType() {
     }
 
     @Override
-    public ScalarFunction makeSubstitute() {
-        ScalarFunction func = null;
+    protected Pipe makePipe() {
+        ArrayList<Pipe> arr = new ArrayList<>(addresses.size());
+        for (Expression address : addresses) {
+            arr.add(Expressions.pipe(address));
+        }
+        return new CIDRMatchFunctionPipe(source(), this, Expressions.pipe(field), arr);
+    }
+
+    @Override
+    public boolean foldable() {
+        if (field.foldable() == false) {
+            return false;
+        }
+        for (Expression address : addresses) {
+            if (address.foldable() == false) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
I believe this exists and would be more succinct.
```suggestion
        return Expressions.foldable(addresses);
```

I think we could also do `Expressions.foldable(children())` in more places.

Reply:
Updated